api.go

```diff
@@ -207,6 +207,17 @@ type (
 		RecordingStatus string `json:"recordingStatus,omitempty"` // ready, waiting
 		RecordingURL    string `json:"recordingUrl,omitempty"`
 		Mp4Url          string `json:"mp4Url,omitempty"`
+
+		Pull *StreamPull `json:"pull,omitempty"`
+	}
+
+	StreamPull struct {
+		Source   string            `json:"source,omitempty"`
+		Headers  map[string]string `json:"headers,omitempty"`
+		Location struct {
+			Lat float64 `json:"lat,omitempty"`
+			Lon float64 `json:"lon,omitempty"`
+		} `json:"location"`
```

Comment:
A pattern I like to use is to (only) use `omitempty` (and pointer if necessary) when the value is optional. So WDYT of?
```suggestion
		Source   string            `json:"source"`
		Headers  map[string]string `json:"headers,omitempty"`
		Location *struct {
			Lat float64 `json:"lat"`
			Lon float64 `json:"lon"`
		} `json:"location,omitempty"`
```
(`lat` and `lon` are optional rn in the schema, but they shouldn't I'll fix it)